### ISSUE TITLE
Feature request: support for duress-passwords

### ISSUE THREAD

**Issue by UnitedMarsupials-zz** [cyrusimap/cyrus-imapd#1831]
A user may find himself under pressure to allow a (potentially) hostile party access to his e-mail. This is already happening on some border-crossings, but criminals may be doing the same.

Though it may be possible for some people to simply maintain multiple accounts, the pressuring entity may already know the person's e-mail address and insist on seeing the contents of the mailboxes _there_.

It would be very useful for people anticipating such a situation to establish additional passwords _for the same account_ and configure the IMAP-server to make some mailboxes unavailable depending on which password was entered.

In addition to the actual duress situations, this new feature could be used to access one's mail from transient locations -- when you wish to check for new mail, but do not want your entire collection of messages downloaded/cached on a computer in an Internet-cafe -- or even casually observed by the cafe's proprietor and other guests.

**Comment by elliefm** [comment]
The difficulty I see with a feature like this is that in order to be useful, no-one can know it exists, in which case it's not useful, because no-one knows it exists.

Consider: if your hostile party knows your mail server supports duress passwords, and they don't find what they're looking for the first time, they can just whack you with a proverbial $5 wrench until you use your _real_ password -- even if you're absolutely innocent and were actually using your real password all along...

**Comment by UnitedMarsupials** [comment]
> if they don't find what they're looking for the first time

Yes, the feature may not help a target of specific investigation -- where the investigators _know_ the target has something, they just need to access its content. But, when they merely _suspect_, the target may be able to protect his privacy, if only in part...

**Comment by elliefm** [comment]
That still only works if they don't know that your mail server supports duress passwords.  If they merely _suspect_ you of anything, but also know you might not be entering your full-access password, out comes the wrench.  Observe too that in the current climate, _suspicion_ may be founded on nothing more substantial than _target isn't very white_.  

I don't think this is appropriate as a feature for Cyrus itself.  It's more a thing an administrator of a small, low profile mail service might quietly bolt on themselves if they find the risk profile appropriate for their user base -- but should definitely be a deliberate, considered choice, not a plug and play/zero thought option.

**Comment by UnitedMarsupials** [comment]
> That still only works if they don't know that your mail server supports duress passwords

I don't believe, that's relevant. But, if you think it is, all the more reasons for _all_ servers to support duress passwords. Like encryption, the feature needs to become so ubiquitous, merely having it enabled will not be indicative of anything.

>  If they merely suspect you of anything, but also know you might not be entering your full-access password

That's a giant "if" -- how would they know?

> in the current climate, suspicion may be founded on nothing more substantial than _target isn't very white_.

While it remains to be seen, what the _current_ Administration's score will be, [In the last fiscal year of the _previous_ Administration, the incidence of such demands for passwords has _quintupled_](http://www.usnews.com/news/us/articles/2017-02-18/electronic-media-searches-at-border-crossings-raise-worry). Whatever your politics, if the reasons for suspicion are, indeed, superficial, the officials will have no reasons to suspect the target of entering a special password.

Perhaps more importantly, this is not limited to interactions with officials. Someone may have their phone configured to _always_ use the special, less-privileged password to access the same account, that's accessed with a different "real" password from desktop at a secure location.

Likewise, the special password may be used to access the account from Internet-cafes and the like.

> It's more a thing an administrator of a small, low profile mail service might quietly bolt on

If Cyrus supports this, an administrator can turn it on. "Bolting it on" is not, unfortunately, an option -- most administrators these days aren't programmers _at all_. And even those, who are (or used to be) will have a hard time figuring out, where to begin.

> not a plug and play/zero thought option.

I disagree completely. It should be plug-and-play -- and on by default server-wide. Whether to actually use the feature -- and which mailboxes to hide -- will be up to individual users.

**Comment by rjbs** [comment]
:robot: I'm closing this issue by automation.

This issue has been sitting untouched for over eight years.  We're trying to get a good handle on our open issues and pull requests, so that we can reliably use them as a backlog of things we hope to work on.  If you are confident that this issue should be reconsidered, reopen it and explain why.

Remember, though, that there are only so many developers working on Cyrus, and we'd like to focus on the most important and/or low-hanging work.

Before you reopen this issue, please check that this issue is still present in the latest release of Cyrus, or the main development branch. An issue is considerably more likely to get resolved if you can provide a pull request which reproduces the issue in a Cassandane test.
